server/container_create.go

```diff
@@ -265,15 +265,28 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 	specgen.SetProcessGID(gid)
 	specgen.AddProcessAdditionalGid(gid)
 
-	for _, group := range addGroups {
-		specgen.AddProcessAdditionalGid(group)
-	}
+	supplementalGroupsPolicy := sc.GetSupplementalGroupsPolicy()
+
+	switch supplementalGroupsPolicy {
+	case types.SupplementalGroupsPolicy_Merge:
+		// Add groups from /etc/passwd and SupplementalGroups defined
+		// in security context.
+		for _, group := range addGroups {
+			specgen.AddProcessAdditionalGid(group)
+		}
+		for _, group := range sc.SupplementalGroups {
+			specgen.AddProcessAdditionalGid(uint32(group))
+		}
+	case types.SupplementalGroupsPolicy_Strict:
```

Comment:
@everpeace in the strict case are we supposed to also add the [gid to supplemental groups](https://github.com/cri-o/cri-o/pull/8268/files#diff-1ec6bb9cf833fd4ded46ce1feb6440f1d7532294e0445623f2f2e0bb2726ea63R266)?

Reply:
@haircommander YES. And, it is MUST. In Linux, the primary gid should always be included/duplicated in supplementary groups. It was reported as CVEs in 2022 in various container runtimes(podmain, CRI-O, containerd). 

For short, if the primary gid is not included in supplementary groups, negative group permission can be bypassed.

See this article for details: https://www.benthamsgaze.org/2022/08/22/vulnerability-in-linux-containers-investigation-and-mitigation/